lib/apiservers/engine/backends/system.go

```diff
@@ -240,7 +241,8 @@ func (s *System) UnsubscribeFromEvents(chan interface{}) {
 }
 
 func (s *System) AuthenticateToRegistry(ctx context.Context, authConfig *types.AuthConfig) (string, string, error) {
-	return "", "", fmt.Errorf("%s does not implement System.AuthenticateToRegistry", ProductName())
+	defer trace.End(trace.Begin(""))
+	return RegistryService.Auth(authConfig, dockerversion.DockerUserAgent(ctx))
```

Comment:
Ok, I'll remove the WIP tag.  Re-implementing the docker daemon was never the point.  Given the short period to do this, it was the most expedient method.  Most of this time was spent on discovery -- how does login works, how does it affect pull, how to setup harbor (secure, and insecure), how does this work with docker.io.  The work on the personality server is a little different from the rest of VIC in that there is always some time spent on discovery of how docker actually works, and as we all see, their documentation doesn't fully detail everything in one place.